### PR TITLE
[chore] [exporter/datasetexporter] Simplify trace exporter, get rid of the client side aggregation code

### PR DESCRIPTION
**Pull request by tomaz-s1**
This pull request updates trace exporter code and gets rid of client side aggregation / populating of `services`,``span_count`` and ``error_count`` attributes.

This client side aggregation is not needed anymore since it now happens on the server side.

It's worth noting that this client side aggregation was meant as a quick short term workaround / hack early in the development cycle (special thanks to @martin-majlis-s1 for implementing this workaround so quickly on the client side).

It was never meant as a long term solution since it had multiple edge cases and problems associated with it - it would only really work correctly / as expected in case all the spans which belong to the same trace are part of the same batch received by the plugin. And of course that won't be true in many scenarios - different spans are routed to different collectors, long running spans which are not part of the same batch, no guarantees how batches are assembled, etc.

As part of this change, `traces.aggregate` and `traces.max_wait` config options have also been made obsolete / redundant.

Those config options have also been removed - since the plugin is still an alpha (some breaking changes are expected) we still have the luxury of being able to do that.

Removing this code also makes the plugin code much simpler now - and who doesn't like removing code and simplifying things (less thing to maintain and worry about edge cases, etc.) :)

### CODE REVIEW

File: .chloggen/dataset-remove-obsolete-trace-client-side-aggregation.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (tomaz-s1) [inline .chloggen/dataset-remove-obsolete-trace-client-side-aggregation.yaml:6]: I labeled it as breaking even though plugin is still in alpha stage and some breaking changes can be expected in alpha stage. Let me know if this should be changed.

**Review by martin-majlis-s1 — APPROVED** [review]
Great work! Thanks @tomaz-s1. :)

**Comment by tomaz-s1** [comment]
@atoulme Please let me know if there is anything I can do / help with, with regards to the review process. Thanks.

**Review by atoulme — APPROVED** [review]
LGTM. It's definitely ok to have breaking changes while in alpha.

**Review by zdaratom-s1 — APPROVED** [review]
LGTM

**Pull request merged**